### ISSUE TITLE
Documentation provides invalid SSL certificate

### ISSUE THREAD

**Issue by fuglede** [cython/cython#3059]
tl;dr: If you want to support HTTPS on docs.cython.org, doing so should boil down to clicking around a bit on the admin dashboard.

Currently, https://docs.cython.org serves an invalid certificate (valid only for readthedocs.io), but SSL on custom domains in Read the Docs should work more or less out-of-the-box. According to [the documentation](https://docs.readthedocs.io/en/stable/custom_domains.html#custom-domain-ssl):

> By default, when you setup a custom domain to host documentation at Read the Docs, we will attempt to provision a domain validated SSL certificate for the domain. This service is generously provided by Cloudflare.

From that same page, it would seem that one needs to use readthedocs.io (iwith no subdomain) in the CNAME record to make use of this functionality though:

> Some older setups configured a CNAME record pointing to readthedocs.org or another variation. While these continue to resolve, they do not yet allow us to acquire SSL certificates for those domains.

And this is probably the reason it fails for cython:

```
$ dig docs.cython.org | grep CNAME
docs.cython.org.        1635    IN      CNAME   cython.readthedocs.io.
```

So presumably, if this functionality is something you would want to make use of (which would be nifty), it seems that all it takes is an update to the CNAME record and a bit of clicking around:

>  Point the CNAME to readthedocs.io, with no subdomain, and re-request a certificate by saving the domain in the project admin
